src/runtime/task/mod.rs

```diff
@@ -44,7 +44,16 @@ pub(crate) const DEFAULT_INLINE_TASKS: usize = 16;
 /// identify tasks in failing Shuttle tests. A task's [Tag] can be set with the
 /// [set_tag_for_current_task](crate::current::set_tag_for_current_task) function. Newly spawned
 /// threads and futures inherit the tag of their parent at spawn time.
-pub trait Tag: Debug {}
+pub trait Tag: Debug {
+    /// Return the tag as `Any`, typically so that it can be downcast to a known concrete type
+    fn as_any(&self) -> &dyn Any;
+}
+
+impl<T> Tag for T where T: Debug + Any {
+    fn as_any(&self) -> &dyn std::any::Any {
```

Comment:
Why the full import `std::any::Any` when you already did a `use std::any::Any` above?